www/js/profile/profile.js

```diff
@@ -1,7 +1,80 @@
 'use strict';
 
-angular.module('main').controller('ProfileCtrl', ['$scope', '$state', '$rootScope', 'AuthService', function ($scope, $state, $rootScope, AuthService) {
+angular.module('main').controller('ProfileCtrl', ['$scope', '$state', '$rootScope', 'AuthService', '$cordovaCamera', function ($scope, $state, $rootScope, AuthService, $cordovaCamera) {
+
+    $scope.passwordShown = false;
+    $scope.showPassword = function(){
+      $scope.passwordShown = !$scope.passwordShown;
+    };
+
+    function cleanForm(form){
+      form.$setPristine();
+      form.$setUntouched();
+    }
+
+    $scope.submit = {
+      email: function(form){
+        if(form.$dirty){
+          AuthService.changeEmail($scope.newProfile.email)
+          .then(function(res){
+            $scope.profile.email = $scope.newProfile.email;
+            cleanForm(form);
+          })
+          .catch(function(err){
+            console.log(err);
+            cleanForm(form);
+          });
+
+        } else cleanForm(form);
+
+      },
+      password: function(form){
+        AuthService.changePassword($scope.newProfile.newPassword)
+        .then(function(res){
+          alert(res);
+          $scope.passwordShown = false;
+          $scope.newProfile.newPassword = null;
+          cleanForm(form);
+        })
+        .catch(function(err){
+          console.log(err);
+          cleanForm(form);
+        });
+      },
+      name: function(form){
+        $scope.profile.name = $scope.newProfile.name;
+        cleanForm(form);
+      },
+      phone: function(form){
+        $scope.profile.phone = $scope.newProfile.phone;
+        cleanForm(form);
+      }
+    };
```

Comment:
Why is `$scope.submit` an object with different methods on it, rather than just a single function that submits everythin on the form?